datastore.go

```diff
@@ -81,9 +81,26 @@ var ErrBatchUnsupported = errors.New("this datastore does not support batching")
 // implement to leverage type safety checks.
 type ThreadSafeDatastore interface {
 	Datastore
+
 	IsThreadSafe()
 }
 
+// CheckedDatastore is an interface that should be implemented by datastores
+// which may need checking on-disk data integrity.
+type CheckedDatastore interface {
+	Datastore
+
+	Check() error
```

Comment:
Maybe `Scrub()`? We also want to support datastores that can do error correction (where an error is returned iff the datastore is unrecoverable).

An open question is how do deal with partial failures. It would nice to be able to recover as much data as possible.

Reply:
`Check` would be invoked by [`ipfs repo fsck`](https://github.com/ipfs/go-ipfs/blob/master/core/commands/repo.go#L211), which for now only removes lock files.

`Scrub()` does make sense (it could be wired up to `ipfs repo verify`, to keep `repo fsck` fast). Does putting this into separate interface SGTY (keeping check for stuff like lock-files)?

Reply:
Got it. Yes, having a separate interface sounds good.

Just to make sure, I assume `Check()` will find and fix any errors related to daemon crashes (i.e., it doesn't just blindly remove the lock file)?

---

Side note: we should get rid of `ipfs repo fsck`; IPFS should do any recovery necessary on startup automatically.